react/ImageList.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { createContext, useContext } from 'react'
+import { defineMessages } from 'react-intl'
+import { useDevice } from 'vtex.device-detector'
+
+import Image from './Image'
+
+interface Props {
+  images: {
+    link?: {
+      url: string
+      noFollow: boolean
+      openNewTab: boolean
+      title: string
+    }
+    image: string
+    mobileImage: string
+    description: string
+  }[]
+  height: number
+}
+
+const ImageListStateContext = createContext<JSX.Element[] | undefined>(
+  undefined
+)
+
+const ImageList: StorefrontFunctionComponent<Props> = ({
+  images,
+  height = 420,
+  children,
+}) => {
+  const { isMobile } = useDevice()
+  const imageListContent = images.map(
+    ({ link, image, mobileImage, description }, idx) => (
+      <Image
+        key={idx}
+        src={isMobile && mobileImage ? mobileImage : image}
+        alt={description}
+        link={link}
+        maxHeight={height}
+        width="100%"
+      />
+    )
+  )
+
+  return (
+    <ImageListStateContext.Provider value={imageListContent}>
+      {children}
+    </ImageListStateContext.Provider>
+  )
+}
+
+export function useImageListState() {
+  const context = useContext(ImageListStateContext)
+  return context
+}
+
+const messages = defineMessages({
+  title: { id: 'admin/editor.image-list.title', defaultMessage: '' },
+  description: {
+    id: 'admin/editor.image-list.description',
+    defaultMessage: '',
+  },
+  imagesImageTitle: {
+    id: 'admin/editor.image-list.images.image.title',
+    defaultMessage: '',
+  },
+  imagesMobileImageTitle: {
+    id: 'admin/editor.image-list.images.mobileImage.title',
+    defaultMessage: '',
+  },
+  imagesImageDescription: {
+    id: 'admin/editor.image-list.images.description.title',
+    defaultMessage: '',
+  },
+  imagesImageLinkUrl: {
+    id: 'admin/editor.image-list.images.link.url.title',
+    defaultMessage: '',
+  },
+  imagesImageLinkOpenNewTab: {
```

Comment:
You are not using some of those strings. Are they really necessary?

Reply:
That's a bit weird, but these are necessary because they are being used at the `contentSchema`, and since there's no `context.json`, they do need to be defined here so that `messages` can detect them.

Reply:
wooow crazy